test/integration/clientset_test.go

```diff
@@ -378,13 +580,242 @@ func TestClusterServiceClassClient(t *testing.T) {
+		return fmt.Errorf("should have same externalName: %q, got %q", name, serviceClasses.Items[0].Spec.ExternalName)
+	}
+
+	serviceClasses, err = serviceClassClient.List(metav1.ListOptions{FieldSelector: "spec.externalName==" + "crap"})
+	if err != nil {
+		return fmt.Errorf("error listing service classes (%s)", err)
+	}
+	if 0 != len(serviceClasses.Items) {
+		return fmt.Errorf("should have zero ClusterServiceClasses, had %v ClusterServiceClasses : %+v", len(serviceClasses.Items), serviceClasses.Items)
+	}
+
+	serviceClasses, err = serviceClassClient.List(metav1.ListOptions{FieldSelector: "spec.clusterServiceBrokerName=" + "test-broker"})
+	if err != nil {
+		return fmt.Errorf("error listing service classes (%s)", err)
+	}
+	if 2 != len(serviceClasses.Items) {
+		return fmt.Errorf("should have two ClusterServiceClasses, had %v ClusterServiceClasses", len(serviceClasses.Items))
+	}
+
+	serviceClasses, err = serviceClassClient.List(metav1.ListOptions{FieldSelector: "metadata.name=" + name})
+	if err != nil {
+		return fmt.Errorf("error listing ClusterServiceClasses: %v", err)
+	}
+
+	if 1 != len(serviceClasses.Items) {
+		return fmt.Errorf("should have exactly one ClusterServiceClass, had %v ClusterServiceClasses", len(serviceClasses.Items))
+	}
+
+	err = serviceClassClient.Delete(name, &metav1.DeleteOptions{})
+	if nil != err {
+		return fmt.Errorf("serviceclass should be deleted (%s)", err)
+	}
+
+	serviceClassDeleted, err := serviceClassClient.Get(name, metav1.GetOptions{})
+	if nil == err {
+		return fmt.Errorf("serviceclass should be deleted (%v)", serviceClassDeleted)
+	}
+
+	err = serviceClassClient.Delete(sc2Name, &metav1.DeleteOptions{})
+	if nil != err {
+		return fmt.Errorf("serviceclass should be deleted (%s)", err)
+	}
+	return nil
+}
+
+// TestClusterServiceClassClient exercises the ClusterServiceClass client.
+func TestNamespacedServiceClassClient(t *testing.T) {
+	rootTestFunc := func(sType server.StorageType) func(t *testing.T) {
+		return func(t *testing.T) {
+			const name = "test-serviceclass"
+			const namespace = "test-namespace"
+			resetFeaturesFunc, err := enableNamespacedResources()
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer resetFeaturesFunc()
+			client, _, shutdownServer := getFreshApiserverAndClient(t, sType.String(), func() runtime.Object {
+				return &servicecatalog.ClusterServiceClass{}
+			})
+			defer shutdownServer()
+
+			if err := testClusterServiceClassClient(sType, client, name); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+	// TODO: Fix this for CRD.
+	// https://github.com/kubernetes-incubator/service-catalog/issues/1256
+	//	for _, sType := range storageTypes {
+	//		if !t.Run(sType.String(), rootTestFunc(sType)) {
+	//			t.Errorf("%q test failed", sType)
+	//		}
+	//	}
+	//	for _, sType := range storageTypes {
+	//		if !t.Run(sType.String(), rootTestFunc(sType)) {
+	//			t.Errorf("%q test failed", sType)
+	//		}
+	//	}
+	sType := server.StorageTypeEtcd
```

Comment:
if this is copy and paste of another test, it should be fixed up as well and remove the CRD comments.

Reply:
Fixed up and removed comments.